src/devices/Ak8963/Ak8963.cs

```diff
@@ -0,0 +1,275 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.IO;
+using System.Numerics;
+using System.Threading;
+
+namespace Iot.Device.Ak8963
+{
+    /// <summary>
+    /// AK8963 class implementing a magnetometer
+    /// </summary>
+    public class Ak8963 : IDisposable
+    {
+        private I2cDevice _i2cDevice;
+        private bool _autoDispose;
+        private MeasurementMode _measurementMode;
+        private OutputBitMode _outputBitMode;
+        private bool _selfTest = false;
+        private Vector3 _magnetometerBias = Vector3.Zero;
+        private Ak8963Interface _ak8963Interface;
+
+        public const byte DefaultI2cAddress = 0x0C;
+
+        public Ak8963(I2cDevice i2CDevice, bool autoDispose = true) : this(i2CDevice, new Ak8963I2c(), autoDispose)
+        { }
+
+        public Ak8963(I2cDevice i2CDevice, Ak8963Interface ak8963Interface, bool autoDispose = true)
+        {
+            _i2cDevice = i2CDevice;
+            _autoDispose = autoDispose;
```

Comment:
@krwq and I chatted offline about this, it is ok to always auto-dispose for I2c so this is not required.

Reply:
ok, will remove